src/elexmodel/handlers/data/CombinedData.py

```diff
@@ -139,13 +139,20 @@ def get_nonreporting_units(self, percent_reporting_threshold, features_to_normal
             # fixed effects manually.
             for expanded_fixed_effect in self.expanded_fixed_effects:
                 if expanded_fixed_effect not in nonreporting_units.columns:
-                    missing_expanded_fixed_effects[expanded_fixed_effect] = [0]
-            missing_expanded_fixed_effects_df = pd.DataFrame(missing_expanded_fixed_effects)
+                    missing_expanded_fixed_effects[expanded_fixed_effect] = 0
+
+            missing_expanded_fixed_effects_df = pd.DataFrame(missing_expanded_fixed_effects, index=[0])
             # if we use this method to add the missing expanded fixed effects because doing it manually
             # can throw a fragmentation warning when there are many missing fixed effects.
-            nonreporting_units = pd.concat([nonreporting_units, missing_expanded_fixed_effects_df], axis=1)
+            # we have to fillna because all other that don't have index 0 will have NaNs for the missing
+            # expanded fixed effect columns
+            nonreporting_units = pd.concat([nonreporting_units, missing_expanded_fixed_effects_df], axis=1).fillna(
+                missing_expanded_fixed_effects
```

Comment:
try the cleaner solution here? Making expanded_fixed_effect_df just a df of 0s, where you get the column names only in the 141 loop

Reply:
thanks for the suggestion! that made the whole thing cleaner and I could get rid of the code below this that dealt with the empty row when `missing_expanded_effects_df` was empty.